08/03/2018 21:17:25.0169	[raft_python_callback.py write_to_logger 60]  169 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:init_state_functions_handler, LINE:28
MESSAGE:start
-----------------------------

08/03/2018 21:17:25.0183	[raft_python_callback.py write_to_logger 60]  183 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:init_state_functions_handler, LINE:70
MESSAGE:end
-----------------------------

08/03/2018 21:17:25.0183	[raft_python_callback.py write_to_logger 60]  183 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:init_multicast_message, LINE:81
MESSAGE:init multicast socket
-----------------------------

08/03/2018 21:17:25.0184	[raft_python_callback.py write_to_logger 60]  184 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:init_multicast_message, LINE:88
MESSAGE:multicast was opened successfully
-----------------------------

08/03/2018 21:17:25.0184	[raft_python_callback.py write_to_logger 60]  184 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:13
MESSAGE:first time
-----------------------------

08/03/2018 21:17:25.0184	[raft_python_callback.py write_to_logger 60]  184 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:create_alarm_timer, LINE:42
MESSAGE:create new alarm timer
VARIABLES:  sharedRaftData.raft_state.current_state=0  timeout=2321 
-----------------------------

08/03/2018 21:17:25.0184	[raft_python_callback.py write_to_logger 60]  184 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=leader_id 
-----------------------------

08/03/2018 21:17:25.0184	[raft_python_callback.py write_to_logger 60]  184 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 
-----------------------------

08/03/2018 21:17:25.0186	[raft_python_callback.py write_to_logger 60]  186 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=commit_idx 
-----------------------------

08/03/2018 21:17:25.0186	[raft_python_callback.py write_to_logger 60]  186 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 
-----------------------------

08/03/2018 21:17:25.0187	[raft_python_callback.py write_to_logger 60]  187 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=applied_last_idx 
-----------------------------

08/03/2018 21:17:25.0187	[raft_python_callback.py write_to_logger 60]  187 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 
-----------------------------

08/03/2018 21:17:25.0189	[raft_python_callback.py write_to_logger 60]  189 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 
-----------------------------

08/03/2018 21:17:25.0189	[raft_python_callback.py write_to_logger 60]  189 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-3 
-----------------------------

08/03/2018 21:17:25.0192	[raft_python_callback.py write_to_logger 60]  192 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 
-----------------------------

08/03/2018 21:17:25.0192	[raft_python_callback.py write_to_logger 60]  192 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 
-----------------------------

08/03/2018 21:17:25.0194	[raft_python_callback.py write_to_logger 60]  194 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:init_raft, LINE:152
MESSAGE:init succeed
-----------------------------

08/03/2018 21:17:25.0194	[raft_python_callback.py write_to_logger 60]  194 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:192
MESSAGE:start raft...
VARIABLES:  raft_ip=230.1.1.10 
-----------------------------

08/03/2018 21:17:25.0194	[raft_python_callback.py write_to_logger 60]  194 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:194
MESSAGE:start raft...
VARIABLES:  raft_port=12345  server_id=1  members_num=2  leader_timeout=500 
-----------------------------

08/03/2018 21:17:25.0194	[raft_python_callback.py write_to_logger 60]  194 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:199
MESSAGE:run raft...
-----------------------------

08/03/2018 21:17:25.0194	[raft_python_callback.py write_to_logger 60]  194 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:131
MESSAGE:try to read message
-----------------------------

08/03/2018 21:17:26.0431	[raft_python_callback.py write_to_logger 60]  431 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:155
MESSAGE:msg was successfully read
VARIABLES:  message_size=20 
-----------------------------

08/03/2018 21:17:26.0432	[raft_python_callback.py write_to_logger 60]  432 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:15
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=1 
-----------------------------

08/03/2018 21:17:26.0432	[raft_python_callback.py write_to_logger 60]  432 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:167
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=0  node_message->event=8  is_relevant_rv=1  node_message->term=53 
-----------------------------

08/03/2018 21:17:26.0432	[raft_python_callback.py write_to_logger 60]  432 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:131
MESSAGE:try to read message
-----------------------------

08/03/2018 21:17:26.0432	[raft_python_callback.py write_to_logger 60]  432 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=8  sharedRaftData.raft_state.current_state=0 
-----------------------------

08/03/2018 21:17:26.0432	[raft_python_callback.py write_to_logger 60]  432 root [DEBUG] 
FILE:./C/Code/follower_handlers.c, FUNCTION:follower_vote_req_handler, LINE:153
MESSAGE:follower get vote request msg
VARIABLES:  sharedRaftData.raft_state.term=0  sharedRaftData.raft_state.did_I_vote=0  node->term=53  node->message_sent_by=2 
-----------------------------

08/03/2018 21:17:26.0432	[raft_python_callback.py write_to_logger 60]  432 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 
-----------------------------

08/03/2018 21:17:26.0432	[raft_python_callback.py write_to_logger 60]  432 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=53 
-----------------------------

08/03/2018 21:17:26.0433	[raft_python_callback.py write_to_logger 60]  433 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=7 
-----------------------------

08/03/2018 21:17:26.0433	[raft_python_callback.py write_to_logger 60]  433 root [DEBUG] 
FILE:./C/Code/follower_handlers.c, FUNCTION:follower_vote_req_handler, LINE:171
MESSAGE:follower sending vote msg
-----------------------------

08/03/2018 21:17:26.0433	[raft_python_callback.py write_to_logger 60]  433 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 
-----------------------------

08/03/2018 21:17:26.0434	[raft_python_callback.py write_to_logger 60]  434 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:155
MESSAGE:msg was successfully read
VARIABLES:  message_size=20 
-----------------------------

08/03/2018 21:17:26.0434	[raft_python_callback.py write_to_logger 60]  434 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:15
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=1 
-----------------------------

08/03/2018 21:17:26.0434	[raft_python_callback.py write_to_logger 60]  434 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:131
MESSAGE:try to read message
-----------------------------

08/03/2018 21:17:26.0434	[raft_python_callback.py write_to_logger 60]  434 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:118
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20 
-----------------------------

